frontend/src/js/actions/workspaces/copyItem.ts

```diff
@@ -0,0 +1,46 @@
+import { copyItem as copyItemApi } from '../../services/WorkspaceApi';
+import {ThunkAction} from "redux-thunk";
+import {GiantState} from "../../types/redux/GiantState";
+import {AppAction, AppActionType, WorkspacesAction} from "../../types/redux/GiantActions";
+import {getWorkspace} from "./getWorkspace";
+
+export function copyItems(
+    workspaceId: string,
+    itemIds: string[],
+    newWorkspaceId?: string,
+    newParentId?: string
+): ThunkAction<void, GiantState, null, WorkspacesAction | AppAction> {
+    return dispatch => {
+        for (const itemId of itemIds) {
+            if (itemId !== newParentId) {
+                console.log("dispatching")
+                dispatch(copyItem(workspaceId, itemId, newWorkspaceId, newParentId));
+            }
+        }
+    };
+}
+
+export function copyItem(
+    workspaceId: string,
+    itemId: string,
+    newWorkspaceId?: string,
+    newParentId?: string
+): ThunkAction<void, GiantState, null, WorkspacesAction | AppAction> {
+    console.log("copying")
+    return dispatch => {
+        return copyItemApi(workspaceId, itemId, newWorkspaceId, newParentId)
+            .then(() => {
+                dispatch(getWorkspace(workspaceId));
+            })
+            .catch(error => dispatch(() => errorCopyingItem(error)));
+    };
+}
```

Comment:
I may be missing something here, but the consumer of this copyItems function is not providing newParentId argument. Could this argument be needed for some future work?

Reply:
You're right - it's not currently used but it could be used in future to support 'copy to folder within the same workspace' functionality. I have added a comment explaining this